### PR TITLE
[types/ember] queryParam types

### PR DESCRIPTION
**Pull request by lifeart**
If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: 
https://github.com/emberjs/ember.js/pull/16701
https://github.com/emberjs/ember.js/blob/6fc89cdf13124d88b8ae6adf99bb02a8c0cdf508/packages/ember-routing/lib/system/router.js#L733

### CODE REVIEW

**Comment by alexlafroscia** [comment]
It seems like `queryParams` on a Controller can also just be a single string, as well as an array of strings or objects (at least, I found an example of this in the docs).

There is also a `scope` and `as` property on the `queryParams` object, both of type `string`, that might be worth adding now as well in the interest of making the definition more accurate.

File: types/ember/index.d.ts [hunk 1]
```diff
@@ -738,7 +739,7 @@ declare module 'ember' {
             replaceRoute(name: string, ...args: any[]): void;
             transitionToRoute(name: string, ...args: any[]): void;
             model: any;
-            queryParams: string[] | Array<{ [key: string]: { type: string } }>;
+            queryParams: string[] | Array<{ [key: string]: { type: QueryParamTypes } }>;
```
Comment (alexlafroscia) [inline types/ember/index.d.ts]: It seems like `queryParams` on a Controller can also just be a single string, as well as an array of strings or objects (at least, I found an example of this in the docs).

There is also a `scope` and `as` property on the `queryParams` object, both of type `string`, that might be worth adding now as well in the interest of making the definition more accurate.

**Review by alexlafroscia — CHANGES_REQUESTED** [review]

File: types/ember/index.d.ts (hunk 1 above)
Comment (lifeart) [reply types/ember/index.d.ts]: Done, https://github.com/emberjs/ember.js/blob/4790fb207c13bdb259451bd5236d572950040ca7/packages/ember-routing/lib/system/route.js#L200

File: types/ember/index.d.ts [hunk 2]
```diff
@@ -111,6 +111,8 @@ declare module 'ember' {
         | 'render'
         | 'afterRender'
         | 'destroy';
+    type QueryParamTypes = 'boolean' | 'number' | 'array' | 'string';
```
Comment (alexlafroscia) [inline types/ember/index.d.ts]: Just wondering, where is the source for these being the available "types"?

**Review by alexlafroscia — APPROVED** [review]

File: types/ember/index.d.ts (hunk 2 above)
Comment (alexlafroscia) [reply types/ember/index.d.ts]: Oh never mind, I now see the links you put in the original description.

**Pull request merged**